ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -367,6 +379,178 @@ public Map apply(@Nullable String fqdn) {
+                .put("description", description)
+                .put("global", global)
+                .put("notification_type", notificationType)
+                .put("alert_states", toArray(alertStates))
+                .put("properties", ImmutableMap.of(
+                        "ambari.dispatch.recipients", formatAmbariAlertsNotificationRecipients(ambariDispatchRecipients),
+                        "mail.smtp.host", mailSmtpHost,
+                        "mail.smtp.port", mailSmtpPort,
+                        "mail.smtp.from", mailSmtpFrom,
+                        "mail.smtp.auth", mailSmtpAuth
+                ))
+                .build();
+
+        updateAlertTarget(registeredAlertNotifications.get(name), mapAlertNotificationsConfigWithDefaults(ambariAlertsConfigFromProperties, alertTargetRequest));
+    }
+
+    private Map<String, Object> mapAlertNotificationsConfigWithDefaults(Map<String, Object> defaultArgs, Map<String, Object> ambariAlertNotifications) {
+        Map<String, Object> result = MutableMap.of();
+
+        result.putAll(defaultArgs);
+        for (Map.Entry config: ambariAlertNotifications.entrySet()) {
+            if (config.getValue() instanceof Map && CollectionUtils.exists(((Map) config.getValue()).values(),NotNullPredicate.INSTANCE)) {
+                Map<String, Object> subresult =  MutableMap.of();
+                if (result.containsKey(config.getKey())) {
+                    subresult.putAll((Map<String, Object>) result.get(config.getKey()));
+                }
+                for (Object subconfig: ((Map) config.getValue()).entrySet()) {
+                    subresult.put((String) ((Map.Entry) subconfig).getKey(), ((Map.Entry) subconfig).getValue());
+                }
+                result.put((String) config.getKey(), subresult);
+            } else {
+                Object value = config.getValue() instanceof Map ? ((Map) config.getValue()).keySet().toArray()[0] : config.getValue();
+                result.put((String) config.getKey(), value);
+            }
+        }
+
+        return result;
+    }
+
+    public Map<String, Integer> listAlertTargets() {
+        AlertTargets alertTargets = restAdapter.create(AlertTargetEndpoint.class).listAlertNotifications();
+
+        Map<String, Integer> result = MutableMap.of();
+        for (Map<String, Object> item: alertTargets.getItems()) {
+            result.put((String) ((Map) item.get("AlertTarget")).get("name"), ((Double) ((Map) item.get("AlertTarget")).get("id")).intValue());
+        }
+
+        return result;
+    }
+
+    public void updateAlertTarget(Integer targetId, Map<String, Object> ambariAlertNotifications) {
+        ImmutableMap<String, Map<String, Object>> alertTargetRequest = ImmutableMap.of("AlertTarget", ambariAlertNotifications);
+
+        restAdapter.create(AlertTargetEndpoint.class).editAlertNotification(targetId, alertTargetRequest);
+    }
+
+    private Map<String, Object> getAmbariAlertsConfigFromProperties() {
```

Comment:
There are default values for the notification parameters, that are set in brooklyn.properties, but they should be merged and/or overwritten by the configs in the yaml. This function gets the properties and merges them. May be it's good to change the name or separate the merge in different function.